space-before-function-paren

aochsner opened the pull request:
Adding option InsertSpaceBeforeFunctionParenthesis
Should be optional
Typically used to support http://eslint.org/docs/rules/space-before-function-paren

Fixes #12234

<!--
Thank you for submitting a pull request!

Here's a checklist you might find useful.
[X ] There is an associated issue that is labelled
  'Bug' or 'Accepting PRs' or is in the Community milestone
[ X] Code is up-to-date with the `master` branch
[X ] You've successfully run `jake runtests` locally
[ X] You've signed the CLA
[X ] There are new or updated unit tests validating the change

Refer to CONTRIBUTING.MD for more details.
  https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md
-->

Fixes #

msftclas commented:
Hi __@aochsner__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

aochsner commented:
Realized that this test will fail... any suggestions as to how to fix?  Guessing that it is conflicting w/ SpaceAfterFunctionInFuncDecl

`/////*3*/var anonymousWithoutSpace = function() {};`
`goTo.marker('3');`
`verify.currentLineContentIs('var anonymousWithoutSpace = function () { };');`

aochsner commented:
wrt anon functions looks like there's a different toggle:
InsertSpaceAfterFunctionKeywordForAnonymousFunctions

aochsner commented:
Okay.  I added a couple more test cases for InsertSpaceBeforeFunctionParenthesis.  Found InsertSpaceAfterFunctionKeywordForAnonymousFuctions to cover that case.  Added InsertSpaceAfterConstructor to handle that case.  That should cover the main cases that http://eslint.org/docs/rules/space-before-function-paren enforces.

Let me know what else should be done.  Thanks!
